modules/rosa-hcp-sharing-vpc-creation-and-sharing.adoc

```diff
@@ -0,0 +1,364 @@
+    },
+    {
+      "Sid" : "VPCEndpointWithCondition",
+      "Effect" : "Allow",
+      "Action" : [
+        "ec2:CreateVpcEndpoint"
+      ],
+      "Resource" : [
+        "arn:aws:ec2:*:*:vpc-endpoint/*"
+      ],
+      "Condition" : {
+        "StringEquals" : {
+          "aws:RequestTag/red-hat-managed" : "true"
+        }
+      }
+    },
+    {
+      "Sid" : "VPCEndpointResourceTagCondition",
+      "Effect" : "Allow",
+      "Action" : [
+        "ec2:CreateVpcEndpoint"
+      ],
+      "Resource" : [
+        "arn:aws:ec2:*:*:security-group*/*"
+      ],
+      "Condition" : {
+        "StringEquals" : {
+          "aws:ResourceTag/red-hat-managed" : "true"
+        }
+      }
+    },
+    {
+      "Sid" : "VPCEndpointNoCondition",
+      "Effect" : "Allow",
+      "Action" : [
+        "ec2:CreateVpcEndpoint"
+      ],
+      "Resource" : [
+        "arn:aws:ec2:*:*:vpc/*",
+        "arn:aws:ec2:*:*:subnet/*",
+        "arn:aws:ec2:*:*:route-table/*"
+      ]
+    },
+    {
+      "Sid" : "ManageVPCEndpointWithCondition",
+      "Effect" : "Allow",
+      "Action" : [
+        "ec2:ModifyVpcEndpoint",
+        "ec2:DeleteVpcEndpoints"
+      ],
+      "Resource" : [
+        "arn:aws:ec2:*:*:vpc-endpoint/*"
+      ],
+      "Condition" : {
+        "StringEquals" : {
+          "aws:ResourceTag/red-hat-managed" : "true"
+        }
+      }
+    },
+    {
+      "Sid" : "ModifyVPCEndpoingNoCondition",
+      "Effect" : "Allow",
+      "Action" : [
+        "ec2:ModifyVpcEndpoint"
+      ],
+      "Resource" : [
+        "arn:aws:ec2:*:*:subnet/*"
+      ]
+    },
+    {
+      "Sid" : "CreateTagsRestrictedActions",
+      "Effect" : "Allow",
+      "Action" : [
+        "ec2:CreateTags"
+      ],
+      "Resource" : [
+        "arn:aws:ec2:*:*:vpc-endpoint/*",
+        "arn:aws:ec2:*:*:security-group/*"
+      ],
+      "Condition" : {
+        "StringEquals" : {
+          "ec2:CreateAction" : [
+            "CreateVpcEndpoint",
+            "CreateSecurityGroup"
+          ]
+        }
+      }
+    }
+  ]
+}
+----
++
+.. Create the policy in AWS:
++
+[source,terminal]
+----
+$ aws iam create-policy \
+    --policy-name VPCEPolicy \
+    --policy-document file:///tmp/vpce-role.json
+----
++
+You will attach this policy to a role necessary for the shared VPC permissions.
++
+.. Create a custom trust policy file that grants permission to assume roles:
++
+[source,terminal]
+----
+$ cat <<EOF > /tmp/vpce-role.json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": "arn:aws:iam::<Account-ID>:root"  <1>
+            },
+            "Action": "sts:AssumeRole"
+        }
+    ]
+}
+EOF
+----
++
+--
+<1> The principal is reduced in scope after the *Cluster Creator* creates the necessary cluster roles. On creation, you must create a root user placeholder by using the *Cluster Creator's* AWS account ID as `arn:aws:iam::{Account}:root`.
```

Comment:
During the initial creation of the principal, we should use the *VPC Owner* AWS account ID instead of the *Cluster Creator* AWS account ID. This is only a temporary placeholder and we should better explain that this is simply a placeholder.